### PR TITLE
fix: Add Service Account Creation for Artifact Registry GCP

### PR DESCRIPTION
**Pull request by hbitoun-aneo**
After the test that occurs the 18/08, we figure that Artifact registry need a service account to be created to add it to the KMS that will encrypt the data. This creation is optionnal if there is already a service account for this service.

### CODE REVIEW

**Comment by hbitoun-aneo** [comment]
When a KMS is created in GCP, Artifact Registry needs a service account that has the rights to encrypt or decrypt the data with the specified KMS. This is done by adding encrypt/decrypt rights to the KMS for the service account otherwise the Artifact Registry will not be able to use the specified KMS.

**Comment by lemaitre-aneo** [comment]
> When a KMS is created in GCP, Artifact Registry needs a service account that has the rights to encrypt or decrypt the data with the specified KMS. That is done by adding encrypt/decrypt rights to the KMS for the service account otherwise the Artifact Registry will not be able to use the specified KMS. [Doc link](https://cloud.google.com/artifact-registry/docs/cmek#create)

My point is: the Service account you've created is not linked in any way to the artifact registry. You should configure the Artifact registry to use the newly created service account.

File: container-registry/gcp/artifact-registry/variables.tf [hunk 1]
```diff
@@ -35,8 +35,11 @@ variable "immutable_tags" {
   default     = true
 }
 
-variable "iam_bindings" {
+variable "iam_roles" {
   description = "Assign role on the repository for a list of users"
   type        = map(list(string))
```
Comment (lemaitre-aneo) [inline container-registry/gcp/artifact-registry/variables.tf:40]:
```suggestion
  type        = map(set(string))
```

**Review by lemaitre-aneo — COMMENTED** [review]
With a terraform format, this will be good to go

File: container-registry/gcp/artifact-registry/variables.tf [hunk 2]
```diff
@@ -1,5 +1,5 @@
 variable "kms_key_name" {
-  description = "KMS key name to encrypt GCP repositories"
+  description = "KMS key name to encrypt GCP repositories. Has the form: projects/{{my-project}}/locations/{{my-region}}/keyRings/{{my-kr}}/cryptoKeys/{{my-key}}"
```
Comment (lemaitre-aneo) [inline container-registry/gcp/artifact-registry/variables.tf:2]: This looks like an id, not a name

**Review by lemaitre-aneo — APPROVED** [review]

File: container-registry/gcp/artifact-registry/variables.tf (hunk 2 above)
Comment (hbitoun-aneo) [reply container-registry/gcp/artifact-registry/variables.tf:2]: yes it is the id of the kms, do I have to update the name of the variable ?

Comment (ftchawe-aneo) [reply container-registry/gcp/artifact-registry/variables.tf:2]: Yes you can update the name of the variable

**Review by ftchawe-aneo — APPROVED** [review]

**Review by lzianekhodja — APPROVED** [review]

**Pull request merged**